tests/test_basic.py

```diff
@@ -201,3 +201,86 @@ def test_pop_save_load(tmp_path):
     assert recs.score.iloc[0] == counts.max()
     # the 10 most popular should be the same
     assert all(counts.index[:10] == recs.item[:10])
+
+
+def test_random():
```

Comment:
Would it be easy enough to split this in to 2-3 test functions so it isn't just one big long one?